crates/sui-core/src/authority/execution_time_estimator.rs

```diff
@@ -74,16 +76,19 @@ impl ExecutionTimeObserver {
         let mut observer = Self {
             epoch_store: Arc::downgrade(epoch_store),
             consensus_adapter,
-            local_observations: HashMap::new(),
+            local_observations: LruCache::new(
+                NonZero::new(lru_cache_size).expect("lru_cache_size must be non-zero"),
```

Comment:
JMO: if you need it to always be nonzero here, the argument to ExecutionTimeObserver::spawn should take a `NonZero`